### PR TITLE
Fix torch median to average the two middle values for even-length inputs

### PR DESCRIPTION
**Pull request by MarcosAsh**
## Summary

`keras.ops.median` returns the wrong value on the torch backend for any reduction over an even number of elements. `torch.median` returns the lower of the two middle values, whereas numpy (which `keras.ops` mirrors) and the jax/tensorflow backends average them.

```python
import numpy as np
from keras import ops
# torch backend
ops.median(ops.convert_to_tensor(np.array([1., 2., 3., 4.])))  # 2.0  (wrong)
# numpy / jax / tensorflow backends, and np.median: 2.5
```

This was wrong on every code path (full reduction, single axis, multiple axes, with and without `keepdims`).

The fix delegates to `quantile(x, q=0.5, ...)`, whose default `method="linear"` averages the two middle values, matching numpy. This mirrors how `nanmedian` in the same module is already implemented (it delegates to `nanquantile(q=0.5)`), and lets the multi-axis / `keepdims` handling reuse the tested `quantile` path instead of a separate implementation.

The existing correctness test only used `[[1, 2, 3], [3, 2, 1]]`, where the two middle values of the flattened array are both `2` (so the lower value coincidentally equals the average) and the `axis=1` reductions are odd-length. No assertion ever exercised an even-length reduction whose middle values differ. This PR adds an even-length case.

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!
